src/Algebra/Construct/Add/Point.agda

```diff
@@ -0,0 +1,97 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Definition of algebraic structures we get from adding a disjoint point
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Algebra.Construct.Add.Point where
+
+open import Algebra.Bundles
+open import Algebra.Core using (Op₂)
+import Algebra.Definitions as Definitions
+import Algebra.Structures as Structures
+open import Data.Maybe.Relation.Binary.Pointwise
+open import Data.Product using (_,_)
+open import Level using (Level; _⊔_)
+open import Relation.Binary.Core
+open import Relation.Binary.Definitions
+import Relation.Binary.Structures as Rel
+open import Relation.Nullary.Construct.Add.Point
+
+private
+  variable
+    a ℓ : Level
+
+module _ {A : Set a} {_≈_ : Rel A ℓ}{refl-≈ : Reflexive _≈_} {op : Op₂ A} where
+  open Definitions {A = Pointed A} (Pointwise _≈_)
+  private
+    module Underlying = Definitions {A = A} _≈_
+
+  lift-≈ : ∀ {x y : A} → x ≈ y → Pointwise _≈_ [ x ] [ y ]
+  lift-≈ = just
+
+  lift-op : Pointed A → Pointed A → Pointed A
+  lift-op [ p ] [ q ] = [ op p q ]
+  lift-op [ p ] ∙     = [ p ]
+  lift-op ∙     [ q ] = [ q ]
+  lift-op ∙     ∙     = ∙
+
+  lift-op-cong₂ : Underlying.Congruent₂ op → Congruent₂ lift-op
+  lift-op-cong₂ R-cong (just eq-l) (just eq-r) = lift-≈ (R-cong eq-l eq-r)
+  lift-op-cong₂ R-cong (just eq) nothing = lift-≈ eq
+  lift-op-cong₂ R-cong nothing (just eq) = lift-≈ eq
+  lift-op-cong₂ R-cong nothing nothing = refl refl-≈
+
+  lift-op-assoc : Underlying.Associative op → Associative lift-op
+  lift-op-assoc assoc [ p ] [ q ] [ r ] = lift-≈ (assoc p q r)
+  lift-op-assoc _     [ p ] [ q ] ∙     = refl refl-≈
+  lift-op-assoc _     [ p ] ∙     [ r ] = refl refl-≈
+  lift-op-assoc _     [ p ] ∙     ∙     = refl refl-≈
+  lift-op-assoc _     ∙     [ r ] [ q ] = refl refl-≈
+  lift-op-assoc _     ∙     [ q ] ∙     = refl refl-≈
+  lift-op-assoc _     ∙     ∙     [ r ] = refl refl-≈
+  lift-op-assoc _     ∙     ∙     ∙     = refl refl-≈
+
+  lift-op-identityˡ : LeftIdentity ∙ lift-op
+  lift-op-identityˡ [ p ] = refl refl-≈
+  lift-op-identityˡ ∙     = refl refl-≈
+
+  lift-op-identityʳ : RightIdentity ∙ lift-op
+  lift-op-identityʳ [ p ] = refl refl-≈
+  lift-op-identityʳ ∙     = refl refl-≈
+
+  lift-op-identity : Identity ∙ lift-op
+  lift-op-identity = lift-op-identityˡ , lift-op-identityʳ
+
+  open Structures {A = Pointed A} (Pointwise _≈_)
+
+  module _ {equiv : Rel.IsEquivalence _≈_}{op-cong₂ : Underlying.Congruent₂ op} where
+    lift-op-isMagma : IsMagma lift-op
```

Comment:
These proofs should be taken out from under the anonymous modules above and take the form:
 - `IsMagma → IsMagma`
 - `IsSemigroup → IsSemigroup`
 - `IsSemigroup → IsMonoid`

and 
 - `Magma → Magma`
 - `Semigroup → Semigroup`
 - `Semigroup → Monoid`

Reply:
Gave this a go but not sure if the structure is exactly what you had in mind since the `Is*` proofs are still parameterized.

Reply:
Yup, exactly what I had in mind!